x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/terms/index.tsx

```diff
@@ -357,22 +360,43 @@ export const termsOperation: OperationDefinition<TermsIndexPatternColumn, 'field
         if ('format' in newParams && newDataType !== 'number') {
           delete newParams.format;
         }
-        const mainField = indexPattern.getFieldByName(fields[0]);
+        const mainField = indexPattern.getFieldByName(sourcefield);
         if (!supportsRarityRanking(mainField) && newParams.orderBy.type === 'rare') {
           newParams.orderBy = { type: 'alphabetical' };
         }
+        // in single field mode, allow the automatic switch of the function to
+        // the most appropriate one
+        if (fields.length === 1) {
+          const newFieldOp = operationSupportMatrix.operationByField[sourcefield]
```

Comment:
This shouldn't just check for the first op, but whether `terms` is in the list or not. Right now switching to a number field while in "Top values" will switch to "Intervals" even though top values on a number field is a valid thing to do